lib/board.rb

```diff
@@ -0,0 +1,51 @@
+
+class Board
+  attr_accessor :cells, :turn_count
+
+  def initialize
+    @cells = Array.new(9, " ")
+    @turn_count = 0
+  end
+
+  def display
+    puts "\e[H\e[2J"
+    puts " #{cells[0]} |" + " #{cells[1]} |" + " #{cells[2]} \n-----------\n"
+    puts " #{cells[3]} |" + " #{cells[4]} |" + " #{cells[5]} \n-----------\n"
+    puts " #{cells[6]} |" + " #{cells[7]} |" + " #{cells[8]} \n\n"
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def position(move)
+    @cells[move.to_i - 1]
+  end
+
+  def full?
+    cells.none?{|c| c == " "}
```

Comment:
Excellent choice of iterator, though [`#all?`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) might be slightly more semantic.